src/Stan/Hie/Debug.hs

```diff
@@ -66,4 +70,14 @@ instance Show ModuleName where
     show = moduleNameString
 
 instance Show Name where
-    show = nameStableString
+    show nm =
+        if isExternalName nm
+        then show $ toNameMeta nm
+        else nameStableString nm
+      where
+        toNameMeta :: Name -> NameMeta
+        toNameMeta name =
+            let nameMetaName = toText $ occNameString $ nameOccName name
+                nameMetaModuleName = fromGhcModule $ nameModule name
+                nameMetaPackage = show @Text $ moduleUnitId $ nameModule name
+            in NameMeta{..}
```

Comment:
Brilliant idea! 💡

Reply:
This should help with debugging :slightly_smiling_face: